internal/host/statemachine.go

```diff
@@ -255,16 +257,69 @@ func NewHostStateMachine(sm stateswitch.StateMachine, th *transitionHandler) sta
 		PostTransition:   th.PostUnbindHost,
 	})
 
-	// Reclaim host
+	// ReclaimHost when installed moves to Reclaiming
 	sm.AddTransition(stateswitch.TransitionRule{
 		TransitionType: TransitionTypeReclaimHost,
 		SourceStates: []stateswitch.State{
 			stateswitch.State(models.HostStatusInstalled),
+			stateswitch.State(models.HostStatusAddedToExistingCluster),
 		},
 		DestinationState: stateswitch.State(models.HostStatusReclaiming),
 		PostTransition:   th.PostUnbindHost,
 	})
 
+	sm.AddTransition(stateswitch.TransitionRule{
+		TransitionType: TransitionTypeRebootingForReclaim,
+		SourceStates: []stateswitch.State{
+			stateswitch.State(models.HostStatusReclaiming),
+		},
+		DestinationState: stateswitch.State(models.HostStatusReclaimingRebooting),
+	})
+
+	sm.AddTransition(stateswitch.TransitionRule{
+		TransitionType: TransitionTypeReclaimFailed,
+		SourceStates: []stateswitch.State{
+			stateswitch.State(models.HostStatusReclaiming),
+			stateswitch.State(models.HostStatusReclaimingRebooting),
```

Comment:
Once we are in HostStatusReclaimingRebooting, how can a failure occur ?  Isn't it just timeout ?

Reply:
The agent could fail to `chroot` or not have required permissions to run reboot.

Reply:
OK